### PR TITLE
Wipe user

### PR DESCRIPTION
**Pull request by metallicalfa2**
We need the ability to wipe all the data for a particular user. The PR has following changes:

- allow users to delete data for all shares (online or offline)
- shouldn't be able to insert old shares in tkey. This is more of a sanity change (changes to catchup to latest shares)
- corresponding tests

### CODE REVIEW

**Comment by metallicalfa2** [comment]
@TrGiLong Here's the follow up PR. Please add your comments here.

**Review by kanishkatn — APPROVED** [review]
lgtm!

**Review by YZhenY — APPROVED** [review]
lgtm

**Comment by YZhenY** [comment]
maybe a stronger warning on the function (even on the on the console)

**Comment by metallicalfa2** [comment]
> maybe a stronger warning on the function (even on the on the console)

i will prefix the function with CRITICAL

File: packages/core/src/core.ts [hunk 1]
```diff
@@ -1242,6 +1236,54 @@ class ThresholdKey implements ITKey {
     };
   }
 
+  getAllShareStoresForLatestPolynomial(): Array<ShareStore> {
```
Comment (chaitanyapotti) [inline packages/core/src/core.ts:1239]: please use ShareStore[] syntax instead of Array<ShareStore>

**Pull request merged**